presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java

```diff
@@ -76,7 +76,7 @@
 
     private final QueryMonitor queryMonitor;
 
-    private final List<WeakReference<Driver>> drivers = new CopyOnWriteArrayList<>();
+    private final Set<WeakReference<Driver>> drivers = Sets.newSetFromMap(new ConcurrentHashMap<>());
```

Comment:
Also add an explanation of why these are weak references and not strong references, since it looks like their lifecycle exactly mirrors the lifecycle of the strong reference in `DriverSplitRunner`